nautobot/utilities/views.py

```diff
@@ -151,3 +186,1062 @@ def get_return_url(self, request, obj=None):
+        model = self.queryset.model
+
+        # Are we deleting *all* objects in the queryset or just a selected subset?
+        if request.POST.get("_all"):
+            if self.bulk_delete_filterset is not None:
+                pk_list = [obj.pk for obj in self.bulk_delete_filterset(request.GET, model.objects.only("pk")).qs]
+            else:
+                pk_list = model.objects.values_list("pk", flat=True)
+        else:
+            pk_list = request.POST.getlist("pk")
+
+        form_cls = self.get_form_for_bulk_delete()
+
+        if "_confirm" in request.POST:
+            form = form_cls(request.POST)
+            if form.is_valid():
+                logger.debug("Form validation was successful")
+
+                # Delete objects
+                queryset = self.queryset.filter(pk__in=pk_list)
+                try:
+                    deleted_count = queryset.delete()[1][model._meta.label]
+                except ProtectedError as e:
+                    logger.info("Caught ProtectedError while attempting to delete objects")
+                    handle_protectederror(queryset, request, e)
+                    return redirect(self.get_return_url(request))
+
+                msg = "Deleted {} {}".format(deleted_count, model._meta.verbose_name_plural)
+                logger.info(msg)
+                messages.success(request, msg)
+                return redirect(self.get_return_url(request))
+
+            else:
+                logger.debug("Form validation failed")
+
+        else:
+            form = form_cls(
+                initial={
+                    "pk": pk_list,
+                    "return_url": self.get_return_url(request),
+                }
+            )
+
+        # Retrieve objects being deleted
+        table = self.table(self.queryset.filter(pk__in=pk_list), orderable=False)
+        if not table.rows:
+            messages.warning(
+                request,
+                "No {} were selected for deletion.".format(model._meta.verbose_name_plural),
+            )
+            return redirect(self.get_return_url(request))
+
+        context = {
+            "form": form,
+            "obj_type_plural": model._meta.verbose_name_plural,
+            "table": table,
+            "return_url": self.get_return_url(request),
+        }
+        context.update(self.get_extra_context(request, "bulk_delete", instance=None))
+        return render(request, self.template_name, context)
+
+    def get_form_for_bulk_delete(self):
+        """
+        Provide a standard bulk delete form if none has been specified for the view
+        """
+
+        class BulkDeleteForm(ConfirmationForm):
+            pk = ModelMultipleChoiceField(queryset=self.queryset, widget=MultipleHiddenInput)
+
+        if self.bulk_delete_form:
+            return self.bulk_delete_form
+
+        return BulkDeleteForm
+
+
+#
+# Views
+#
+
+
+class ObjectView(ObjectPermissionRequiredMixin, ObjectDetailViewMixin, View):
```

Comment:
Some comments would be helpful to explain:

- how `utilities.views.ObjectView` (etc.) is similar to / different from `core.views.generic.ObjectView` (etc.)
- when I would want to use `ObjectView` versus when I should use the mixins directly